addons/bus/static/src/js/async_systray.js

```diff
@@ -0,0 +1,145 @@
+odoo.define('bus.systray.AsyncSpinner', function (require) {
+"use strict";
+
+const {bus, qweb, _lt, _t} = require('web.core');
+const SystrayMenu = require('web.SystrayMenu');
+const Widget = require('web.Widget');
+const {AsyncJobService, asyncJobState} = require('bus.AsyncJobService');
+
+const STATE_DATA = {
+    [asyncJobState.CREATED]: {
+        icon: 'fa fa-arrow-circle-right',
+        tooltip: _lt('The task has been scheduled for processing and will start shortly.'),
+    },
+    [asyncJobState.PROCESSING]: {
+        icon: 'fa fa-spinner fa-spin',
+        tooltip: _lt('The task is processing.'),
+    },
+    [asyncJobState.SUCCEEDED]: {
+        icon: 'fa fa-check-circle',
+        tooltip: _lt('The task has been processed, click to resume the execution.'),
+    },
+    [asyncJobState.FAILED]: {
+        icon: 'fa fa-exclamation-circle text-danger',
+        tooltip: _lt('The task failed due to an error, click to reveal it.'),
+    },
+    [asyncJobState.DONE]: {
+        icon: 'fa fa-check-circle text-success',
+        tooltip: _lt('The task has be completely executed, re-click to re-execute it.'),
+    },
+}
```

Comment:
Not gonna lie, I think use of this `enum` is inappropriate. It contains things that should be directly in template.
I also think that use of `typedef` looked smart at first (less typos with hard-coded strings, some autocompletion with IDE), but the code itself is hard to read, so code maintenance is quite poor.